src/EventSubscriber/CreateNewWishlistSubscriber.php

```diff
@@ -108,60 +88,26 @@ public function onKernelResponse(ResponseEvent $event): void
             return;
         }
 
-        if ($event->getRequest()->cookies->has($this->wishlistCookieToken)) {
+        $request = $this->requestStack->getMainRequest();
```

Comment:
If there are more places where you retrieve $this->requestStack->getMainRequest(), it might be better to move the variable to the constructor and create $this->mainRequest there

Reply:
I can do this